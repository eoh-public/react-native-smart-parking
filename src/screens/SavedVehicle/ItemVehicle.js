import React, { memo, useCallback } from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import t from 'i18n';

import { Colors } from '../../configs';
import Text from '../../commons/Text';
import VehicleIconInfo from '../VehicleManagement/components/VehicleInfoIcon';
import { SvgCarPrimary } from '../../../assets/images/SmartParking';

const ItemVehicle = memo((props) => {
  const { car, onPress, showDefault, onUpdateDefault } = props;

  const onPressItem = useCallback(() => {
    if (showDefault) {
      onUpdateDefault && onUpdateDefault(car.id);
    } else {
      onPress && onPress(car);
    }
  }, [showDefault, onUpdateDefault, car, onPress]);

  const renderNameCar = (name = car.name, seats = car.seats) => {
    return `${name ? name : ''} ${name && seats ? ' - ' : ''} ${
      seats ? `${seats} ${t('seats')}` : ''
    }`;
  };

  return (
    <TouchableOpacity
      style={styles.container}
      activeOpacity={0.4}
      onPress={onPressItem}
    >
      <SvgCarPrimary />
      <View style={styles.info}>
        <Text type="H4" semibold color={Colors.Gray9}>
          {car.plate_number}
        </Text>
        <Text type="Label" size={12} color={Colors.Gray8}>
          {renderNameCar()}
        </Text>
      </View>
      <VehicleIconInfo style={styles.vehicleContainer} {...props} />
    </TouchableOpacity>
  );
});

export default ItemVehicle;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    paddingHorizontal: 16,
    flexDirection: 'row',
    paddingTop: 16,
  },
  info: {
    flex: 1,
    marginLeft: 16,
    borderBottomWidth: 1,
    borderBottomColor: Colors.Gray4,
    paddingBottom: 16,
  },
  vehicleContainer: {
    position: 'absolute',
    top: 16,
    right: 16,
  },
});
