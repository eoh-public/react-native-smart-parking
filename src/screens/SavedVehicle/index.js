import React, { memo, useCallback, useState } from 'react';
import { StyleSheet, View, FlatList } from 'react-native';
import { getBottomSpace } from 'react-native-iphone-x-helper';
import { useNavigation } from '@react-navigation/native';

import { Colors } from '../../configs';
import { FullLoading } from '../../commons';
import keyExtractor from '../../utils/keyExtrator';
import Routes from '../../utils/Route';

import ItemVehicle from './ItemVehicle';

const SavedVehicle = memo(({ route }) => {
  const {
    parkingId,
    spot_id,
    spot_name,
    cars,
    loadingCars,
    refreshCars,
    onRefreshCars,
    numBookHour,
  } = route.params;
  const { navigate } = useNavigation();
  const [index, setIndex] = useState(-1); //index choose

  const onChoose = useCallback(
    (car) => {
      setIndex(car.id);
      navigate(Routes.SmartParkingParkingAreaDetail, {
        id: parkingId,
        spot_id: spot_id,
        spot_name: spot_name,
        carItem: car,
        unLock: true,
        numBookHour,
      });
    },
    [navigate, numBookHour, parkingId, spot_id, spot_name]
  );

  const renderItem = useCallback(
    ({ item }) => {
      return <ItemVehicle car={item} onPress={onChoose} index={index} />;
    },
    [index, onChoose]
  );

  return (
    <View style={styles.container}>
      {loadingCars ? (
        <FullLoading wrapStyle={styles.container} />
      ) : (
        <FlatList
          data={cars}
          keyExtractor={keyExtractor}
          renderItem={renderItem}
          refreshing={refreshCars}
          onRefresh={onRefreshCars}
        />
      )}
    </View>
  );
});

export default SavedVehicle;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.White,
    paddingBottom: getBottomSpace(),
  },
});
