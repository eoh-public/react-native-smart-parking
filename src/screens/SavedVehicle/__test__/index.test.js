import React from 'react';
import renderer, { act } from 'react-test-renderer';

import SavedVehicle from '../index';

const route = {
  params: {
    cars: [
      {
        background: '',
        id: 155,
        is_default: true,
        name: 'Xe Ben',
        plate_number: '84H-123.45',
        seats: 2,
      },
    ],
  },
};

describe('test saved vehicle container', () => {
  let tree;

  test('render saved vehicle container', () => {
    act(() => {
      tree = renderer.create(<SavedVehicle route={route} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
