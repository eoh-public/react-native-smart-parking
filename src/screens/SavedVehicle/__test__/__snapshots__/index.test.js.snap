// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test saved vehicle container render saved vehicle container 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingBottom": 0,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "background": "",
          "id": 155,
          "is_default": true,
          "name": "Xe Ben",
          "plate_number": "84H-123.45",
          "seats": 2,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <SvgMock />
          <View
            style={
              Object {
                "borderBottomColor": "#E8E8E8",
                "borderBottomWidth": 1,
                "flex": 1,
                "marginLeft": 16,
                "paddingBottom": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#262626",
                    "fontFamily": "SFProDisplay-Semibold",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "textAlign": "left",
                    "textDecorationLine": "none",
                  },
                  Object {},
                ]
              }
            >
              84H-123.45
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#595959",
                    "fontFamily": "SFProDisplay-Regular",
                    "fontSize": 12,
                    "lineHeight": 20,
                    "textAlign": "left",
                    "textDecorationLine": "none",
                  },
                  Object {},
                ]
              }
            >
              Xe Ben  -  2 chỗ
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                },
                Object {
                  "position": "absolute",
                  "right": 16,
                  "top": 16,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#00979D",
                      "fontFamily": "antoutline",
                      "fontSize": 21,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SFProDisplay-Regular",
                      "fontSize": undefined,
                      "lineHeight": undefined,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "fontSize": 10,
                      "lineHeight": 20,
                      "marginTop": 1.5,
                    },
                  ]
                }
              >
                MẶC ĐỊNH
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#F08229",
                    "fontFamily": "antfill",
                    "fontSize": 21,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
